src/test/java/ActionableMessageHelperImplTest.java

```diff
@@ -0,0 +1,138 @@
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import com.google.api.services.gmail.model.Message;
+import com.google.api.services.gmail.model.MessagePart;
+import com.google.api.services.gmail.model.MessagePartHeader;
+import com.google.common.collect.ImmutableList;
+import com.google.sps.exceptions.GmailMessageFormatException;
+import com.google.sps.model.ActionableMessage;
+import com.google.sps.model.ActionableMessageHelper;
+import com.google.sps.model.ActionableMessageHelperImpl;
+import java.util.Collections;
+import java.util.List;
+import org.junit.Assert;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+/** Tests the ActionableMessageHelperImpl and its helper methods. */
+@RunWith(JUnit4.class)
+public class ActionableMessageHelperImplTest {
+  private static final ActionableMessageHelper actionableMessageHelperImpl =
+      new ActionableMessageHelperImpl();
+
+  private static final List<String> LABELS_STARRED = ImmutableList.of("STARRED");
+  private static final List<String> LABELS_IMPORTANT = ImmutableList.of("IMPORTANT");
+  private static final String USER_EMAIL = "example@example.com";
+  private static final String NOT_USER_EMAIL = "fake@fake.com";
+  private static final String TO_HEADER_VALUE_USER_EMAIL = String.format("<%s>", USER_EMAIL);
+  private static final String TO_HEADER_VALUE_NOT_USER_EMAIL =
+      String.format("<%s>", NOT_USER_EMAIL);
+  private static final String TO_HEADER_VALUE_UNDISCLOSED_RECIPIENTS =
+      ActionableMessageHelperImpl.UNDISCLOSED_RECIPIENTS_TO_HEADER;
+  private static final String TO_HEADER_VALUE_TOO_MANY_RECIPIENTS =
+      createHeaderWithNRecipients(
+          ActionableMessageHelperImpl.MAILING_LIST_LOWER_BOUND, TO_HEADER_VALUE_USER_EMAIL, ", ");
+  private static final MessagePartHeader toHeaderUserEmail =
+      new MessagePartHeader().setName("To").setValue(TO_HEADER_VALUE_USER_EMAIL);
+  private static final MessagePartHeader toHeaderNotUserEmail =
+      new MessagePartHeader().setName("To").setValue(TO_HEADER_VALUE_NOT_USER_EMAIL);
+  private static final MessagePartHeader toHeaderUndisclosedRecipients =
+      new MessagePartHeader().setName("To").setValue(TO_HEADER_VALUE_UNDISCLOSED_RECIPIENTS);
+  private static final MessagePartHeader toHeaderTooManyRecipients =
+      new MessagePartHeader().setName("To").setValue(TO_HEADER_VALUE_TOO_MANY_RECIPIENTS);
+
+  private static final Message starredMessage =
+      new Message()
+          .setPayload(new MessagePart().setHeaders(Collections.singletonList(toHeaderNotUserEmail)))
```

Comment:
Very minor comment: `ImmutableList.of(element)` is a way to express a single-item list with less typing. I think the shorter form is actually more readable, because more people are familiar with `ImmutableList.of` construct.

Reply:
100% Agreed - fixing this now! Thanks